File: leetcode/silding_window/1456/imp.cpp
#include <cstddef>

using namespace std;

class Solution {
    public: 
        int maxVowels(string s, int k) {
            size_t currentNum = 0;
            size_t maxNum = 0;

            if (k ==0) {
                return 0;
            }

            for (size_t i = 0 ;i < k; ++i) {
                if (isVowel(s[i])) {
                    currentNum += 1;
                }
            }
            maxNum = currentNum;
            if (maxNum == k) {
                return maxNum;
            }

            for (size_t i = k; i < s.length(); ++i) {
                if (isVowel(s[i])) {
                    ++currentNum;
                }
                if (isVowel(s[i - k])) {
                    --currentNum;
                }
                if (currentNum == k) {
                    return currentNum;
                }
                maxNum = max(maxNum, currentNum);
            }

            return maxNum;
        }

        bool isVowel(char c){
            return c == 'a' ||
 c == 'e' ||
 c == 'i' ||
 c == 'o' ||
 c == 'u'; 
        }
};

int main() {

}

File: leetcode/silding_window/1456/imp.py

class Solution:
    def isVowel(self, c: str) -> bool:
        return c in {'a', 'e', 'i', 'o', 'u'}

    def maxVowels(self, s: str, k: int) -> int:
        if k == 0 or len(s) < k: 
            return 0
        max = 0
        curr = 0

        for i in range(k):
            if self.isVowel(s[i]):
                curr += 1

        max = curr
        if max == k:
            return k

        for i in range(k, len(s)):
            if self.isVowel(s[i]):
                curr += 1
            if (self.isVowel(s[i - k])):
                curr -= 1

            if curr > max:
                max = curr
                if max == k:
                    return max


        return max

